extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java

```diff
@@ -423,73 +422,85 @@ private static void applyOpenshiftResources(OpenShiftClient client, List<HasMeta
+    private static Build startOpenshiftBuild(BuildConfig buildConfig, File binaryFile,
+            OpenshiftConfig openshiftConfig, KubernetesClientBuilder kubernetesClientBuilder) {
+        try (KubernetesClient kubernetesClient = kubernetesClientBuilder.build()) {
+            OpenShiftClient client = toOpenshiftClient(kubernetesClient);
+            try {
+                return client.buildConfigs().withName(buildConfig.getMetadata().getName())
+                        .instantiateBinary()
+                        .withTimeoutInMillis(openshiftConfig.buildTimeout.toMillis())
+                        .fromFile(binaryFile);
+            } catch (Exception e) {
+                Optional<Build> running = buildsOf(client, buildConfig).stream().findFirst();
+                if (running.isPresent()) {
+                    LOG.warn("An exception: '" + e.getMessage()
+                            + " ' occurred while instantiating the build, however the build has been started.");
+                    return running.get();
+                } else {
+                    throw openshiftException(e);
+                }
             }
         }
+    }
+
+    private static void waitForOpenshiftBuild(Build build, OpenshiftConfig openshiftConfig,
+            KubernetesClientBuilder kubernetesClientBuilder) {
 
         while (isNew(build) || isPending(build) || isRunning(build)) {
             final String buildName = build.getMetadata().getName();
-            Build updated = client.builds().withName(buildName).get();
-            if (updated == null) {
-                throw new IllegalStateException("Build:" + build.getMetadata().getName() + " is no longer present!");
-            } else if (updated.getStatus() == null) {
-                throw new IllegalStateException("Build:" + build.getMetadata().getName() + " has no status!");
-            } else if (isNew(updated) || isPending(updated) || isRunning(updated)) {
-                build = updated;
-                try (LogWatch w = client.builds().withName(buildName).withPrettyOutput().watchLog();
-                        Reader reader = new InputStreamReader(w.getOutput())) {
-                    display(reader, openshiftConfig.buildLogLevel);
-                } catch (IOException e) {
-                    // This may happen if the LogWatch is closed while we are still reading.
-                    // We shouldn't let the build fail, so let's log a warning and display last few lines of the log
-                    LOG.warn("Log stream closed, redisplaying last " + LOG_TAIL_SIZE + " entries:");
-                    try {
-                        display(client.builds().withName(buildName).tailingLines(LOG_TAIL_SIZE).getLogReader(),
-                                Logger.Level.WARN);
-                    } catch (IOException ex) {
-                        // Let's ignore this.
+            try (KubernetesClient kubernetesClient = kubernetesClientBuilder.build()) {
```

Comment:
Let's open a new connection every time we iterate (there should not be many times).